File: src/test/resources/InorderTraversal.java
import java.util.*;

import com.jalgoarena.type.TreeNode;

public class Solution {

    public int[] inorderTraversal(TreeNode root) {
        List<Integer> items = new ArrayList<>();
        inorder(root, items);
        return items.stream().mapToInt(i -> i).toArray();
    }

    private void inorder(TreeNode root, List<Integer> items) {
        if (root != null) {
            inorder(root.left, items);
            items.add(root.data);
            inorder(root.right, items);
        }
    }
}

File: src/test/resources/InorderTraversal.kt
import com.jalgoarena.type.TreeNode
import java.util.*
import com.jalgoarena.type.*

class InorderTraversal {

    fun inorderTraversal(root: TreeNode?): IntArray {
        val items = ArrayList<Int>()
        inorder(root, items)
        return items.toIntArray()
    }

    private fun inorder(root: TreeNode?, items: MutableList<Int>) {
        if (root != null) {
            inorder(root.left, items)
            items.add(root.data)
            inorder(root.right, items)
        }
    }
}
